eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -91,14 +92,20 @@ case class NodeParams(nodeKeyManager: NodeKeyManager,
                       maxPaymentAttempts: Int,
                       enableTrampolinePayment: Boolean) {
   val privateKey: Crypto.PrivateKey = nodeKeyManager.nodeKey.privateKey
+
   val nodeId: PublicKey = nodeKeyManager.nodeId
-  val keyPair = KeyPair(nodeId.value, privateKey.value)
 
-  val pluginMessageTags: Set[Int] = pluginParams.flatMap(_.tags).toSet
+  val keyPair: KeyPair = KeyPair(nodeId.value, privateKey.value)
+
+  val pluginMessageTags: Set[Int] = pluginParams.collect { case p: CustomFeaturePlugin => p.messageTags }.toSet.flatten
+
+  def nonStandardIncomingHtlcs: Seq[PostRestartHtlcCleaner.IncomingHtlc] = pluginParams.collect { case p: CustomCommitmentsPlugin => p.getIncomingHtlcs }.flatten
+
+  def nonStandardRelayedOutHtlcs(htlcsIn: Seq[PostRestartHtlcCleaner.IncomingHtlc]): Map[Origin, Set[(ByteVector32, Long)]] = pluginParams.collect { case p: CustomCommitmentsPlugin => p.getHtlcsRelayedOut(htlcsIn) }.flatten.toMap
```

Comment:
This taken from this discussion: https://gitter.im/ACINQ/developers?at=5fa1289c2a3544071500e1b3